### PR TITLE
[MPS] Add regression test for sliced matmul

### LINKED ISSUE #104832: MPS matmul with sliced (strided) out argument produces wrong output, may corrupt memory
### 🐛 Describe the bug

When using `torch.matmul` with an `out` argument that is a view that is not contiguous, the op has no change to the `out` argument instead of computing the matmul correctly. Here is a minimal example that replicates the behavior. It seems to happen consistently across many different sizes (i.e. the `256` here can be changed to something else and it still outputs zeros).

```
import torch

X = torch.randn(256,256).to("mps")
Y = torch.randn(256,256).to("mps")
Z = torch.zeros(256,2*256).to("mps")

torch.matmul(X,Y,out=Z[:,128:(128+256)])
```

Here, `torch.matmul` both outputs all zeros and does not change `Z`. Compare the behavior of the non-MPS code

```
import torch

X = torch.randn(256,256)
Y = torch.randn(256,256)
Z = torch.zeros(256,2*256)

torch.matmul(X,Y,out=Z[:,128:(128+256)])
```

which runs correctly and produces a non-zero output.

After doing this, I sometimes get segfaults in my program, which suggests that this is also corrupting memory, although I am unable to find any code that consistently replicates this segfault behavior.

### Versions

Collecting environment information...
PyTorch version: 2.0.1
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: macOS 13.4 (arm64)
GCC version: Could not collect
Clang version: 14.0.3 (clang-1403.0.22.14.1)
CMake version: version 3.26.3
Libc version: N/A

Python version: 3.9.13 (main, Aug 25 2022, 18:24:45)  [Clang 12.0.0 ] (64-bit runtime)
Python platform: macOS-13.4-arm64-arm-64bit
Is CUDA available: False
CUDA runtime version: No CUDA
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Apple M2 Max

Versions of relevant libraries:
[pip3] numpy==1.24.0
[pip3] pytorch-pretrained-bert==0.6.2
[pip3] torch==2.0.1
[pip3] torchaudio==0.13.1
[pip3] torchdata==0.6.1
[pip3] torchtext==0.15.2
[pip3] torchvision==0.13.1
[conda] htorch                    0.1                      pypi_0    pypi
[conda] numpy                     1.24.0                   pypi_0    pypi
[conda] pytorch-pretrained-bert   0.6.2                    pypi_0    pypi
[conda] torch                     2.0.1                    pypi_0    pypi
[conda] torchaudio                0.13.1                 py39_cpu    pytorch
[conda] torchdata                 0.6.1                    pypi_0    pypi
[conda] torchtext                 0.15.2                   pypi_0    pypi
[conda] torchvision               0.13.1                   pypi_0    pypi

cc @kulinseth @albanD @malfet @DenisVieriu97 @jhavukainen @razarmehr @abhudev

### PR DESCRIPTION
**Pull request by hvaara**
This issue seems to have been fixed in macOS already. This PR only adds a regression test. I'm expecting failure on test runners with macOS 13 and 14.

Fixes #104832

### CODE REVIEW

**Comment by hvaara** [comment]
I would appreciate it if someone could add the `ciflow/mps` label 🙏

**Comment by hvaara** [comment]
@malfet Can you help me with permission for running the tests?

**Review by malfet — APPROVED** [review]

File: test/test_mps.py [hunk 1]
```diff
@@ -680,6 +680,22 @@ def test_large_matmul(self):
...
+        x_mps = x_cpu.detach().clone().to("mps")
+
+        y_cpu = torch.randn(256, 256, device="cpu")
+        y_mps = y_cpu.detach().clone().to("mps")
+
+        out_cpu = torch.zeros(256, 2 * 256, device="cpu")
+        out_mps = torch.zeros(256, 2 * 256, device="mps")
+
+        torch.matmul(x_cpu, y_cpu, out=out_cpu[:, 128:(128+256)])
+        torch.matmul(x_mps, y_mps, out=out_mps[:, 128:(128+256)])
```
Comment (malfet) [inline test/test_mps.py]:
```suggestion
        torch.matmul(x_cpu, y_cpu, out=out_cpu[:, 128:128 + 256])
        torch.matmul(x_mps, y_mps, out=out_mps[:, 128:128 + 256])
```
